otherlibs/stdune/src/bool.ml

```diff
@@ -16,4 +16,6 @@ let of_string s = bool_of_string_opt s
 
 let to_dyn t = Dyn.Bool t
 
-let hash (t : bool) = Poly.hash t
+let hash = function
```

Comment:
Also for this, shouldn't we use a more "uniform" choice rather than 0 and 1.

Reply:
It probably doesn't make much difference either way because there can be at most 2 keys. Any two `int`s are probably equivalent in terms of uniformity.

Reply:
We should really be using an array instead of a hash table for Boolean keys.

Reply:
> We should really be using an array instead of a hash table for Boolean keys

Indeed, but we have no such tables at the moment. I'm only adding this so that our various compound functions that use `Bool.hash` get a faster hash function.